### PR TITLE
EVG-5346 (fix) Perf. discovery - correct ratio calculation

### PR DESCRIPTION
**Pull request by unrealsolver**

### CODE REVIEW

**Comment by bsamek** [comment]
@dalyd, could you take a look?

File: public/static/app/perfdiscovery/perfDiscoveryRatio.js [hunk 1]
```diff
@@ -14,6 +14,13 @@ mciModule.directive('perfDiscoveryRatio', function() {
       // This $watch is required due to known bug with ui-grid
       // https://github.com/angular-ui/ui-grid/issues/4869
       scope.$watch('ratio', function(val) {
+        var val = Math.round(
+          (scope.ratio > 1 ? scope.ratio - 1 : 1 - 1 / scope.ratio) * 100
```
[the remark this answers is not in the document]
Comment (unrealsolver) [reply public/static/app/perfdiscovery/perfDiscoveryRatio.js]: Basically, the formula is following
```python
ratio = lambda a, b: a / b - 1 if a > b else 1 - b / a
```
For example, let's compare 100 to 50, it give us +100%
And vice versa 50 to 100, gives us -100%
Which looks correct to me. More generated examples:
```c
0.25 to 1 is  -300%
0.33 to 1 is -200%
0.50 to 1 is -100%
0.75 to 1 is -33%
1.00 to 1 is 0%
1.33 to 1 is +33%
1.50 to 1 is +50%
2.00 to 1 is +100%
3.00 to 1 is +200%
4.00 to 1 is +300%
```

[the remark this answers is not in the document]
Comment (bsamek) [reply public/static/app/perfdiscovery/perfDiscoveryRatio.js]: Ultimately this is up to @dalyd, but shouldn't your second case be -50%?
```
ratio = lambda base, new: (new - base) / base
```

[the remark this answers is not in the document]
Comment (dalyd) [reply public/static/app/perfdiscovery/perfDiscoveryRatio.js]: The ratio should not be symmetric. In Ruslan's example there, it should be a 100% increase followed by a 50% decrease. Anything else would be too surprising. Subtracting 1 from the current ratio to give percent change is a reasonable change, but we need to communicate it out. 

The current scheme of just reporting the ratio is multiplicatively symmetric. In Ruslan's example we have a ratio of 2 followed by a ratio of 0.5. 2*0.5 = 1, is the identity. 

We should also make sure that the latencies (negative numbers) work properly. So if somthing goes from -100 to -200, that should be 1/2 in the current scheme, or -50% if we subtract out 1.

**Comment by john-m-liu** [comment]
@dalyd are you ok with this merging as is?

**Comment by dalyd** [comment]
Nope. Not okay with merging. See open comment in code.

**Comment by unrealsolver** [comment]
Sounds reasonable. Going to close this then.

**Pull request closed without merging**